include/command.h

```diff
@@ -119,23 +108,36 @@ namespace detail {
 	};
 
 	class task_command : public abstract_command {
-		friend class command_graph;
-
 	  protected:
-		task_command(command_id cid, node_id nid, task_id tid, subrange<3> execution_range)
-		    : abstract_command(cid, nid), tid(tid), execution_range(execution_range) {}
+		task_command(command_id cid, node_id nid, task_id tid) : abstract_command(cid, nid), tid(tid) {}
 
 	  public:
 		task_id get_tid() const { return tid; }
 
+	  private:
+		task_id tid;
+	};
+
+	class horizon_command final : public task_command {
+		friend class command_graph;
+		using task_command::task_command;
```

Comment:
How neat. I wasn't actively aware of this.

Reply:
Re-exporting base-class constructors? Yeah, that is pretty neat!